Makefile

```diff
@@ -227,6 +227,12 @@ ifeq ($(PLATFORM),kubernetes)
 else
 	$(K8S_CLI) apply -f config/registry/local/os -n $(NAMESPACE)
 endif
+#evaluate plugin registry URL to use in other targets
+ifeq ($(PLATFORM),kubernetes)
+export PLUGIN_REGISTRY_URL=http://che-plugin-registry.$(ROUTING_SUFFIX)/v3)
+else
+export PLUGIN_REGISTRY_URL=http://$(shell $(K8S_CLI) get route -n $(NAMESPACE) che-plugin-registry -o=yaml | yq -r '.status.ingress[].host')/v3
+endif
```

Comment:
All lines that aren't part of a rule (since these are unindented they're not part of `deploy_registry`) are evaluated before any rules, so if the plugin registry is not already deployed this will export `PLUGIN_REGISTRY_URL=http:///v3`. E.g. using the Makefile

```makefile
default:        
        @echo default
$(info 'hi')
```
will output
```
❯ make
'hi'
default
```

One (extremely ugly) way to work around this would be to use `eval`:
```makefile
default: set_var
        @echo $(SOME_VAR) # works without export below
        @echo ${SOME_VAR} # works without export below
        @cat myfile.txt | envsubst # works *only* with export below

set_var:
        $(eval export SOME_VAR := "hi")
```

Reply:
We also might still run into an edge-case here, there's a chance the plugin registry route will not have been reconciled across the cluster and so may not have a URL set (I don't recall the exact mechanics here -- maybe it's always set).

Reply:
Looking into the close-timing parts of it: `.status.ingress[].host` gets set once the route is ready, but `.spec.host` gets set when the route is created. If you check too soon after creating the route, there's a chance (I can reproduce it fairly often) that `.status.ingress` will be `null`.